src/shogun/features/DotFeatures.cpp

```diff
@@ -64,6 +64,31 @@ float64_t CDotFeatures::dense_dot_sgvec(int32_t vec_idx1, SGVector<float64_t> ve
 	return dense_dot(vec_idx1, vec2.vector, vec2.vlen);
 }
 
+void CDotFeatures::dense_dot_range_parallel(float64_t* output, int32_t start, int32_t stop, float64_t* alphas, float64_t* vec, int32_t dim, float64_t b)
```

Comment:
dont put parallel in name of this method pls, rather "batch" or so
But I support the idea of putting that in